spot-client/src/spot-remote/ui/components/nav/buttons/audio-mute-button.js

```diff
@@ -0,0 +1,111 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { connect } from 'react-redux';
+
+import {
+    getInMeetingStatus,
+    getOptimisticAudioMuteState,
+    isAudioMutePending,
+    setAudioMute
+} from 'common/app-state';
+
+import NavButton from './../nav-button';
+
+/**
+ * A component for displaying and changing the current audio mute of a Spot-TV.
+ *
+ * @extends React.Component
+ */
+export class AudioMuteButton extends React.Component {
+    static propTypes = {
+        audioMuted: PropTypes.bool,
+        changePending: PropTypes.bool,
+        setAudioMute: PropTypes.func
+    };
+
+    /**
+     * Initializes a new {@code AudioMuteButton} instance.
+     *
+     * @param {Object} props - The read-only properties with which the new
+     * instance is to be initialized.
+     */
+    constructor(props) {
+        super(props);
+
+        this._onToggleAudioMute = this._onToggleAudioMute.bind(this);
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     */
+    render() {
+        const { audioMuted, changePending } = this.props;
+        let label;
+
+        if (changePending) {
+            label = audioMuted ? 'Muting...' : 'Unmuting...';
+        } else {
+            label = audioMuted ? 'Unmute Audio' : 'Mute Audio';
+        }
+
+        return (
+            <NavButton
+                active = { changePending ? !audioMuted : audioMuted }
+                className = { changePending ? 'pending' : '' }
+                iconName = { audioMuted ? 'mic_off' : 'mic' }
+                label = { label }
+                onClick = { this._onToggleAudioMute }
+                qaId = { audioMuted ? 'unmute-audio' : 'mute-audio' } />
+        );
+    }
+
+    /**
+     * Changes the current local audio mute state of a Spot-TV.
+     *
+     * @private
+     * @returns {void}
+     */
+    _onToggleAudioMute() {
+        if (this.props.changePending) {
+            return;
+        }
+
+        this.props.setAudioMute(!this.props.audioMuted);
+    }
+}
+
+/**
+ * Selects parts of the Redux state to pass in with the props of
+ * {@code AudioMuteButton}.
+ *
+ * @param {Object} state - The Redux state.
+ * @private
+ * @returns {Object}
+ */
+function mapStateToProps(state) {
+    const optimisticMuteState = getOptimisticAudioMuteState(state);
```

Comment:
I would probably move the logic about "optimistic state" to inside of the selector. If you'd want to display audio mute state anywhere else in the app this logic would have to be duplicated.